course/enrollment.py

```diff
@@ -210,10 +231,18 @@ def enroll_view(request, course_identifier):
                         pass
             pass
 
+    def email_suffix_matches(email, suffix):
+        # type: (Text, Text) -> bool
+        if suffix.startswith("@"):
+            return email.endswith(suffix)
+        else:
+            return email.endswith("@%s" % suffix) or email.endswith(".%s" % suffix)
```

Comment:
I'd prefer this at the top level, to clarify that it's not using lexical context.

Reply:
Otherwise I like this solution. :)

Reply:
shall I pull them out?